data_cache.py

```
@@ -62,6 +62,45 @@
+                    fd = os.open(str(self._path), os.O_CREAT | os.O_EXCL | os.O_WRONLY)
+                    try:
+                        os.write(fd, str(os.getpid()).encode("utf-8"))
+                    except OSError:
+                        os.close(fd)
+                        self._path.unlink(missing_ok=True)
+                        raise
+                    self._fd = fd
+                    return
+                except FileExistsError:
+                    if time.monotonic() >= deadline:
+                        raise FileLockTimeout(f"Timed out waiting for lock {self._path}")
```

Ruff (0.15.9)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)